app/models/bet.rb

```diff
@@ -2,4 +2,41 @@ class Bet < ApplicationRecord
   has_many :users, through: :user_bets
   has_many :user_bets
   has_many :options, through: :user_bets, source: :bet_option
+
+  def resolve
+    losers.each do |debtor|
+      debt = amount_bet(debtor)
+      win_contributions.each do |contribution|
+        debtor.debts.create(creditor: contribution[:creditor],
+                            amount:   debt * contribution[:percentage])
+      end
+    end
+  end
+
+  def win_option
+    options.winners
+  end
+
+  def winner_total
+    user_bets.winners.map(&:amount_bet).sum
+  end
+
+  def winners
+    users.winners
+  end
+
+  def losers
+    users.losers
+  end
+
+  def amount_bet(user)
+    user_bets.find_by(user: user).amount_bet
+  end
+
+  def win_contributions
+    winners.map do |w|
+      winner_individual = amount_bet(w)
+      { creditor: w, percentage: (winner_individual / winner_total).round(2) }
```

Comment:
This method returns a complex data structure of the form:

```ruby
[
  { creditor:, percentage: },
  { creditor:, percentage: },
  { creditor:, percentage: }
]
```

This feels overly complex and I think you can achieve a similar effect more simply. If you really needed to return such a complicated data structure, it would probably be an indication that you are missing a class to model that data.